enhancements/operator-dependency-resolution.md

```diff
@@ -0,0 +1,216 @@
+
+### Syntax
+
+The dependency list will contain a `type` field for each item to specify what kind of dependency this is. It can be a package type (`olm.package`) meaning this is a dependency for a specific operator. For `olm.package` type, the dependency information should include the `package` name and the `version` of the package in semver format. We use `blang/semver` library for semver parsing (https://github.com/blang/semver). For example, you can specify an exact version such as `0.5.2` or a range of version such as `>0.5.1` (https://github.com/blang/semver#ranges). In addition, the author can specify dependency that is similiar to existing CRD/API-based using proposed `crd` or `api` type and then specify GVK information as how it is done in CSV. This is a path to enable operator authors to consolidate all dependencies (API or explicit) to be in the same place.
+
+Moreover, for explicit operator dependency, a new field named `indexNamespace` is proposed to enable operator authors to choose a specific source where the operator is provided. This concept is similiar to choosing a specific CatalogSource. However, we choose not to include CatalogSource information in the bundle as that information is out of scope of a bundle. Operator authors shouldn't need to know about all CatalogSources that are available in the cluster as they may not be the same across different clusters. By using this `indexNamespace` type, we require changes in pipeline that build CatalogSources to add a new label `indexNamespace` that indicates the "index namespace" in term of provider where all of operators inside belong to. This `indexNamespace` constrain will allow OLM to pick the correct operator when the same operator is included in different CatalogSource.
+
+Here is an example of CatalogSource with `indexNamespace` label:
+
+```
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: catsrc-test
+spec:
+  displayName: CatalogSource Test
+  sourceType: grpc
+  image: docker.io/test/catsrc-test:latest
+  indexNamespace: io.redhat
+```
+
+An example of a `dependencies.yaml` that specifies Prometheus operator and etcd CRD dependencies:
+
+```
+dependencies:
+  - type: olm.package
+    indexNamespace: io.redhat
+    name: prometheus
+    version: >0.27.0
+  - type: crd
+    name: etcdclusters.etcd.database.coreos.com
+    group: etcd.database.coreos.com
+    kind: EtcdCluster
+    version: v1beta2
+```
+
+#### Provided APIs
+
+The two default types of provided APIs (operator name + version and CRD GVK) will be implied just by the registry parsing the yaml provided in the bundle. However, we acknowledge that in the future there may be addition APIs provided by operators that are not as easily discoverable.
```

Comment:
Should this be something like, "there are two default ways dependencies can be satisfied"?

I just mean that name+version is not really a "provided api"

Reply:
I see. I wonder if you have any suggestion on how the syntax for provided APIs look like especially for Service Mesh (Istio) example that you mentioned in our meeting. Should it look like olm.gvk type but then I'm not sure who to specify a range of istio API with that?